src/web/app/pages-session/session-submission-page/session-submission-page.component.ts

```diff
@@ -547,84 +549,63 @@ export class SessionSubmissionPageComponent implements OnInit, AfterViewInit {
-                    questionType: questionSubmissionFormModel.questionType,
-                    responseDetails: recipientSubmissionFormModel.responseDetails,
-                  }).pipe(
-                      tap((resp: FeedbackResponse) => {
-                        recipientSubmissionFormModel.responseId = resp.feedbackResponseId;
-                        recipientSubmissionFormModel.responseDetails = resp.responseDetails;
-                        recipientSubmissionFormModel.recipientIdentifier = resp.recipientIdentifier;
-                      }),
-                      switchMap(() => this.createCommentRequest(recipientSubmissionFormModel)),
-                      catchError((error: ErrorMessageOutput) => {
-                        failToSaveQuestions[questionSubmissionFormModel.questionNumber] = error.error.message;
-                        return of(error);
-                      }),
-                  ));
+            if (!isFeedbackResponseDetailsEmpty) {
+              responseDetailsPerRecipient[recipientSubmissionFormModel.recipientIdentifier] =
+                  recipientSubmissionFormModel.responseDetails;
             }
           });
 
+      savingRequests.push(
+          this.feedbackResponsesService.submitFeedbackResponse(questionSubmissionFormModel.feedbackQuestionId, {
+            intent: this.intent,
+            key: this.regKey,
+            moderatedperson: this.moderatedPerson,
+          }, {
+            responseDetailsPerRecipient,
+            questionType: questionSubmissionFormModel.questionType,
+          }).pipe(
+              tap((resp: FeedbackResponses) => {
+                const responsesMap: Record<string, FeedbackResponse> = {};
+                resp.responses.forEach((response: FeedbackResponse) => {
+                  responsesMap[response.recipientIdentifier] = response;
+                });
+
+                questionSubmissionFormModel.recipientSubmissionForms
+                    .forEach((recipientSubmissionFormModel: FeedbackResponseRecipientSubmissionFormModel) => {
+                      if (responsesMap[recipientSubmissionFormModel.recipientIdentifier]) {
+                        const correspondingResp: FeedbackResponse =
+                            responsesMap[recipientSubmissionFormModel.recipientIdentifier];
+                        recipientSubmissionFormModel.responseId = correspondingResp.feedbackResponseId;
+                        recipientSubmissionFormModel.responseDetails = correspondingResp.responseDetails;
+                        recipientSubmissionFormModel.recipientIdentifier = correspondingResp.recipientIdentifier;
+                      } else {
+                        recipientSubmissionFormModel.responseId = '';
+                        recipientSubmissionFormModel.commentByGiver = undefined;
+                      }
+                    });
+              }),
+              // switchMap(() =>
```

Comment:
I skipped this because I wasn't very sure how this is supposed to work. Will take some time to add this back, @xpdavid if you can help it will be great.